roles/edpm_pre_adoption_validation/molecule/adoption-negative/converge.yml

```diff
@@ -0,0 +1,19 @@
+---
+- name: Converge
+  hosts: all
+  tasks:
+    - name: Converge
+      block:
+        - name: "Include role"
+          ansible.builtin.include_role:
+            name: osp.edpm.edpm_pre_adoption_validation
+
+        - name: "Check execution halted"
+          ansible.builtin.fail:
+            msg: "Execution should stop before this task"
+          register: should_not_run
+      rescue:
+        - name: Asset that role failed
+          ansible.builtin.assert:
+            that:
+              - should_not_run is not defined
```

Comment:
ah yes so the way this si working is 

- name: "Check execution halted" should not run if the validation failed

so the rescue block assert that should_not_run is not defied to asset that the validation failed.

its a little indricet but i don't know of a better way to assert that a role failed while also not causing the playbook to fail.

Reply:
I agree. I tried but failed to found a more direct pattern.